apps/frontend/src/app/5_pages/EarnPage/EarnPage.tsx

```diff
@@ -293,6 +341,40 @@ const EarnPage: FC = () => {
     [],
   );
 
+  useEffect(() => {
+    if (index) {
+      setShowUnderCollateralizedError(false);
+    }
+  }, [index]);
+
+  useEffect(() => {
+    if (index !== 1 || isInMaintenance) {
+      setIsUnderCollateralized(false);
+      return;
+    }
+
+    if (troves?.troves) {
+      const isLowTroveExists = troves.troves.reduce(
+        (acc, { collateral, debt }) => {
+          const collateralRatio = calculateCollateralRatio(
+            Number(collateral),
+            Number(debt),
+            Number(price),
```

Comment:
```suggestion
            decimalic(collateral.toString()),
            decimalic(debt.toString()),
            decimalic(price.toString()),
```

I expect collateral, debt and price to be Decimal type from zero SDK? If so convert it to the string and then use `decimalic` helper to convert to sovryn's Decimal type.

This will prevent floating number loss